lib/ui/bottombar/bottom_navigation_bar_controller.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:farmsmart_flutter/ui/bottombar/tab_navigator.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+class BottomNavigationBarController extends StatefulWidget {
+  final List<TabNavigator> tabs;
+
+  const BottomNavigationBarController({
+    Key key,
+    this.tabs,
+  }) : super(key: key);
+
+  @override
+  _BottomNavigationBarControllerState createState() =>
+      _BottomNavigationBarControllerState();
+}
+
+class _BottomNavigationBarControllerState
+    extends State<BottomNavigationBarController> {
+  int _selectedIndex = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    return WillPopScope(
+      onWillPop: () async => !await widget
```

Comment:
async and !await.... maybe its my dart knowledges lacking, but I don't understand what's happening here, please explain to me :).  its reads like, run this async, but then when running the function it doesn't wait for another async function (it might as well wait, as its already async) I guess I'm not sure what !await does.

Reply:
@LeeMatthewHiggins - I think that the `!` does the negation of the Future<bool> `.maybePop()`. Is not a negation for the await 😅

Reply:
👍